spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -929,6 +929,17 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
       binding: Boolean): Option[Expr] = {
     SQLConf.get
 
+    def toProto(handler: CometExpressionSerde): Option[Expr] = {
+      if (handler.isIncompat && !CometConf.COMET_EXPR_ALLOW_INCOMPATIBLE.get()) {
+        withInfo(
+          expr,
+          s"$expr is not fully compatible with Spark. To enable it anyway, set " +
+            s"${CometConf.COMET_EXPR_ALLOW_INCOMPATIBLE.key}=true")
+        return None
+      }
```

Comment:
We no longer have to repeat this check and error message for each expression, we can simply override the `isCompat` method instead,

Another option here would be to have expressions extend an additional `IncompatExpr` trait, which may be better.